### ISSUE TITLE
[analyzer] completion.getSuggestions2 does not show local variables if called right before variable assignment

### ISSUE THREAD

**Issue by julemand101** [dart-lang/sdk#49385]
Issue reproduced with:
```lang-none
Dart SDK version: 2.17.5 (stable) (Tue Jun 21 11:05:10 2022 +0200) on "windows_x64"
Dart SDK version: 2.18.0-238.0.dev (dev) (Wed Jun 29 13:41:53 2022 -0700) on "windows_x64"
```
With IntelliJ IDEA 2022.1.3 (Community Edition) using Dart plugin version 221.5921.27.

**Problem**
When doing the following code:
```dart
void main() {
  String someVariable = 'A';
  String anotherVariable = 'B';
  
  someVariable = 'C';
}
```
The auto completion seems to ignore locale variables if trying to get suggestions about `someVariable` and `anotherVariable` before `someVariable = 'C';`:
![image](https://user-images.githubusercontent.com/116227/176960555-263f2e1f-b2c5-4912-95bc-6de587dd002a.png)
![image](https://user-images.githubusercontent.com/116227/176960595-08e10686-115d-4547-834a-3a15fc365098.png)

While it works fine if doing it after the variable assignment:
![image](https://user-images.githubusercontent.com/116227/176960747-13ebfb6e-ef6b-424c-b84a-bcda3fb0d8f0.png)

I have also tested with global variables and that also works fine. So it seems to be specific for local variables and it does not matter what variable the assignment are about:
![image](https://user-images.githubusercontent.com/116227/176960921-2bba9ff4-5ac6-42f7-ac8d-510f68123a3d.png)

For verifying this issue is not about IntelliJ, I have tried using `--instrumentation-log-file` against the Analyzer Server spawned though IntelliJ Dart plugin, and I can see it gets suggestions using `completion.getSuggestions2` and does indeed miss the local variables when doing the call at a position before the variable declaration.

I did also test with VS Code with Dart v3.44.0 but it seems it uses Language Server Protocol with the method `textDocument/completion` which does not seem to have the problem when I enable the `--instrumentation-log-file` logging and tries to reproduce the issue. It might also be for different reasons but in the end, I could not reproduce the issue in VS Code.

**Comment by julemand101** [comment]
I can no longer reproduce the issue in 2.19.0-146.0.dev (2.18.0 does have the issue), so I close this issue.

**Comment by scheglov** [comment]
Most probably it was fixed by 92cdc0b380d4af017dbbd3d9d15cbefcd54a6d68.
